### PR TITLE
ADD ability to sync css & head content

### PR DESCRIPTION
**Pull request by robnester-rh**
This commit addresses HACBS-2111 and adds the ability to sync common css & head content between the Hugo & Antora portions of the website. This is achieved with two scripts, `sync-css.js` and `sync-head.js`. Both can be executed independently with `npm run sync-css` & `npm run sync-head`.

The make file has been updated to execute these sync scripts prior to executing the builds of Antora / Hugo.

### CODE REVIEW

**Review by zregvart — COMMENTED** [review]
Not too happy with this approach, I was hoping to get to a point where we don't need any extra scripts. i.e. Hugo builds the CSS and the same CSS is used from Antora. I think this can be done by pointing to the Hugo built CSS files, perhaps disregarding the fingerprinting on the first pass. This seems needlessly complex.

Getting 404s on these resources:
https://adf91341.enterprise-contract.pages.dev/docs/_/css/base.css
https://adf91341.enterprise-contract.pages.dev/docs/_/css/ec-custom.css
https://adf91341.enterprise-contract.pages.dev/docs/_/css/highlight.css

From here: https://adf91341.enterprise-contract.pages.dev/docs/ec/main/

**Comment by robnester-rh** [comment]
> Not too happy with this approach, I was hoping to get to a point where we don't need any extra scripts. i.e. Hugo builds the CSS and the same CSS is used from Antora. I think this can be done by pointing to the Hugo built CSS files, perhaps disregarding the fingerprinting on the first pass. This seems needlessly complex.

My conversation with @simonbaird prompted me to work up a solution that provided a shared base CSS that would be used in both portions, with additional css specific to Hugo or Antora  available to them respectively.

> 
> Getting 404s on these resources: https://adf91341.enterprise-contract.pages.dev/docs/_/css/base.css https://adf91341.enterprise-contract.pages.dev/docs/_/css/ec-custom.css https://adf91341.enterprise-contract.pages.dev/docs/_/css/highlight.css
> 
> From here: https://adf91341.enterprise-contract.pages.dev/docs/ec/main/

I'll check into that.

**Comment by robnester-rh** [comment]
>Getting 404s on these resources:
https://adf91341.enterprise-contract.pages.dev/docs/_/css/base.css
https://adf91341.enterprise-contract.pages.dev/docs/_/css/ec-custom.css
https://adf91341.enterprise-contract.pages.dev/docs/_/css/highlight.css

>From here: https://adf91341.enterprise-contract.pages.dev/docs/ec/main/

404s resolved. My apologies,  I'd intended to remove those links prior to pushing the code.

File: Makefile [hunk 1]
```diff
@@ -3,21 +3,27 @@ help:
 	@grep '[[:space:]]##[[:space:]]' Makefile | sed 's/^\(.*\):.*##\(.*\)$$/#\2\nmake \1\n/'
 
 preview: preview-antora ## Run hugo server for website hacking
+	@npm run sync-css && npm run sync-head
 	@cd website && hugo server --config hugo.toml
```
Comment (simonbaird) [inline Makefile]:
```suggestion

sync-common-content:
	@npm run sync-css && npm run sync-head

preview: sync-common-content preview-antora ## Run hugo server for website hacking
	@cd website && hugo server --config hugo.toml

```

Comment (simonbaird) [reply Makefile]: Better to pull the npm run comands into their own make target.

File: scripts/sync-css.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  .join('\n');
+
+// Prepend the comment to the shared content
+const combinedCSSContentWithComment = cssComment + '\n\n' + combinedCSSContent; + '\n'
+
+// Save the content to Hugo's CSS file
+fs.writeFileSync(hugoCSSPath, combinedCSSContentWithComment, 'utf8');
+
+// Save the content to Antora's CSS file
+fs.writeFileSync(antoraCSSPath, combinedCSSContentWithComment, 'utf8');
```
Comment (simonbaird) [inline scripts/sync-css.js:30]: Hmm, this does feel more complicated than what I was imagining.

File: Makefile (hunk 1 above)
Comment (robnester-rh) [reply Makefile]: ack

File: scripts/sync-css.js (hunk 2 above)
Comment (robnester-rh) [reply scripts/sync-css.js:30]: If preferred, I can update this PR such that all common css lives in `/website/assets/css/ec-common.css` with `/website/assets/css/custom.css` and `/antora/supplemental-ui/css/custom.css` being for platform specific overrides. I can then update the antora head template to link to `/site.main`, but the fingerprint would be an issue due to the current way the website is built (we build Antora, then Hugo, so we won't know the fingerprint).

Comment (simonbaird) [reply scripts/sync-css.js:30]: I was looking at a minimal version and made #70. It's uses (yet more) symlinks, and doesn't do the packing and fingerprinting, but I think maybe it's good enough for now.

**Comment by robnester-rh** [comment]
Closing in lieu of #74

**Pull request closed without merging**